src/main/java/com/oneops/secrets/command/SecretGet.java

```diff
@@ -69,13 +71,30 @@ public void exec() {
   private Path save(String name, byte[] content) throws IOException {
     Path path = Paths.get(name);
     if (path.toFile().exists()) {
-      String in =
-          System.console()
-              .readLine(warn("File " + path + " exists. Do you want to overwrite (y/n)? "));
-      if (in == null || !in.equalsIgnoreCase("y")) {
-        throw new IllegalStateException("Exiting");
+
+      if (null != System.console()) {
+        String in =
+            System.console()
+                .readLine(warn("File " + path + " exists. Do you want to overwrite (y/n)? "));
+        if (in == null || !in.equalsIgnoreCase("y")) {
+          throw new IllegalStateException("Exiting");
+        }
+        return Files.write(path, content, CREATE);
+      } else {
+        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+        String input = "";
+        try {
+          input = reader.readLine();
+        } catch (IOException e) {
+          throw new IllegalStateException("Exiting");
+        }
+        if (input.equalsIgnoreCase("y") || input.equalsIgnoreCase("yes")) {
+          return Files.write(path, content, CREATE);
+        } else {
+          throw new IllegalStateException("Exiting");
+        }
       }
     }
-    return Files.write(path, content, CREATE);
+    throw new IllegalStateException("Exiting");
```

Comment:
Don't throw exception on normal flow. This can be problematic later when we add new condition or refactor the code. As earlier, returns Files.write for normal flow and throw exception for error scenarios.